### PR TITLE
Iplab2 script

### PR DESCRIPTION
**Pull request by rfree-d**
Script to setup IPs on test computers.

(the .cpp change should be merged in other PR first #516)

### CODE REVIEW

**Comment by rfree-d** [comment]
Could you confirm this PR? @leonkakaowiec6   @tx1683

**Comment by rfree-d** [comment]
@tx1683 can you re-review it in the new version? perahsp also a quick ACK if you would fully test exactly this version (details sent in PM) thanks :)

File: src-tools/iplab2-script/iplab2.sh [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+do
+	dev=${ipCARD[ix-1]} # -1 to move to 0-based bash array indexes
+	echo "Your cardnr :$ix is $dev"
+	ix=$((ix+1))
+	ip address flush dev ${dev}
+done
+echo "---------------------"
+echo
+
+[[ "$ipMyBase" -gt 9 ]] && fail "Too big ipMyBase ($ipMyBase)"
```
Comment (rfree-d) [inline src-tools/iplab2-script/iplab2.sh]: also test against -lt 1

File: src-tools/iplab2-script/iplab2.sh [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+	target2=$( echo "$target" | sed -e's/:/ /g')
+	read my_nicnr target_base target_nicnr <<< "$target2" || fail "Parse error in $[target]"
+	[[ "$my_nicnr" =~ ^[0-9]{1,3}$ ]] || fail "Not a valid integer in [$target] ($my_nicnr) my_nicnr"
+	[[ "$target_base" =~ ^[0-9]{1,3}$ ]] || fail "Not a valid integer in [$target] ($target_base) target_base"
+	[[ "$target_nicnr" =~ ^[0-9]{1,3}$ ]] || fail "Not a valid integer in [$target]  ($target_nicnr) target_nicnr"
+
+	[[ "$my_nicnr" -gt 9 ]] && fail "Too big my_nicnr ($my_nicnr)"
+	[[ "$target_base" -gt 9 ]] && fail "Too big target_base ($target_base)"
+	[[ "$target_nicnr" -gt 9 ]] && fail "Too big target_nicnr ($target_nicnr)"
+
```
Comment (rfree-d) [inline src-tools/iplab2-script/iplab2.sh]: 3 times:
also test against -lt 1

File: src-tools/iplab2-script/iplab2.sh [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+	# b1 and b2: if I am base 4, and target is 7 then our common net is .74. sorted, so .47.
+	# this assumes that b1<b2, then build net b1.b2 and subnet c1.c2
```
Comment (rfree-d) [inline src-tools/iplab2-script/iplab2.sh]: and subnet n1.n2 not c1.c2

File: src-tools/iplab2-script/iplab2.sh [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+
+echo "---------------------"
+echo "Your cards (configured in $config_global)"
+ix=1
+for target in "${ipCARD[@]}"
+do
+	dev=${ipCARD[ix-1]} # -1 to move to 0-based bash array indexes
+	echo "Your cardnr :$ix is $dev"
+	ix=$((ix+1))
+	ip address flush dev ${dev}
```
Comment (rfree-d) [inline src-tools/iplab2-script/iplab2.sh]: quote the device name

File: src-tools/iplab2-script/iplab2.sh [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+
+	my_eth="${ipCARD[my_nicnr-1]}"
+	echo "My $my_ip (:$my_nicnr=$my_eth) ---to----> his $target_ip (:$target_nicnr)"
+
+	#set -x
+	ip address flush dev ${my_eth}
+	ip link set dev ${my_eth} up
+	ip addr add "${my_ip}/${ipNetmask}" dev "$my_eth"
+	ifconfig "${my_eth}" mtu 9000
+	#set +x
```
Comment (rfree-d) [inline src-tools/iplab2-script/iplab2.sh]: in all above ip.... commands: in 4 lines
quote the device name

File: src-tools/iplab2-script/iplab2.sh (hunk 1 above)
Comment (rfree-d) [reply src-tools/iplab2-script/iplab2.sh]: :ok:

File: src-tools/iplab2-script/iplab2.sh (hunk 2 above)
Comment (rfree-d) [reply src-tools/iplab2-script/iplab2.sh]: :ok:

File: src-tools/iplab2-script/iplab2.sh (hunk 3 above)
Comment (rfree-d) [reply src-tools/iplab2-script/iplab2.sh]: :ok:

**Comment by rfree-d** [comment]
😱 😱 😱 @tx1683 whoops, you missed possible exploit, but hopefully very unlikelly one (attacker would need to write into root-writable /etc/ config).

Though this is very low risk bug, anyway that config is executed, so if attacker could write there anyway he could run arbitrary code (though that might be worth improving on later)


unquoted $eth could give arbitrary additional argumetns to `ip address ...`  (not sure, perhaps even shell commands?) if /etc/ would be written to, and if then user could sudo this script (a common use case)

seems fixed now

:ok:

**Comment by rfree-d** [comment]
@tx1683 wan to ACK, in current version?

**Comment by tx1683** [comment]
ACK tested -u and -c

**Comment by robertoleksy** [comment]
Concept ACK

**Comment by sdf5** [comment]
Concept ACK

File: src-tools/kpktgen/kpktgen.sh [hunk 6]
```diff
@@ -33,10 +34,12 @@ function prepare_sending() {
 	card="$1"
 	target="$2"
 	threads="$3"
-	send_start="$4"
+	port_low="$4"
+	port_high="$5"
+	send_start="$6"
 	mtu="$5"
```
Comment (robertoleksy) [inline src-tools/kpktgen/kpktgen.sh]: repeat $5 parameter

**Comment by robertoleksy** [comment]
Concept ACK

**Pull request merged**